third_party/purge/views/accessory_purge_varnish.php

```diff
@@ -1,21 +1,38 @@
-<a id="purge_varnish" href="#">Send purge request to Varnish now.</a>
-
+<small style="position:relative; top: -3px;">Enter URL starting with slash (not http etc) or leave blank to purge everything.</small><br>
+<div style="position:relative;">
+<input id="purge_url" type="text" style="width:350px; margin:0 3px 0 0;">
+<button id="purge_varnish" href="#" style="height:100%;position:absolute;">Purge</button>
+</div>
 <script type="text/javascript" charset="utf-8">
 	$(document).ready(function()
 	{
+		$('#purge_url').click(function()
+		{
+			if( ! $(this).val().match(/^\//) ) $(this).val('');
+		});
+		
 		$('#purge_varnish').click(function(event)
 		{
 			event.preventDefault();
 			
-			$link = $(this);
-			$link.html('Sending purge request...');
+			var url = $('#purge_url').val(); 
+			
+			if( url != '')
+			if( ! url.match(/^\/[^\/]/) ) { $('#purge_url').val('URL must start with one "/"'); return; }
```

Comment:
Then you could probably remove that click event above. I guess I'm just thinking of the case where you enter something like "about", and then it warns you to add a slash, then you just have to add a slash to the beginning instead of typing out the whole thing again.